util/lambda/index.js

```diff
@@ -0,0 +1,318 @@
+/* eslint-disable no-useless-catch */
+/* eslint-disable no-unused-vars */
+const JSZip = require("jszip");
+const fs = require("fs");
+const {
+  LambdaClient,
+  CreateFunctionCommand,
+  GetFunctionCommand,
+  InvokeCommand,
+  UpdateFunctionCodeCommand,
+  waitUntilFunctionActiveV2,
+  waitUntilFunctionUpdatedV2
+} = require("@aws-sdk/client-lambda");
+const logger = require("../../logger");
+const stats = require("../stats");
+const {
+  isABufferValue,
+  bufferToString,
+  TRANSFORM_WRAPPER_CODE
+} = require("./utils");
+
+const MAX_WAIT_TIME = parseInt(process.env.MAX_WAIT_TIME || "30", 10);
+const DELAY = parseInt(process.env.DELAY || "2", 10);
+const EVENT_SIZE_LIMIT_IN_MB =
+  parseInt(process.env.EVENT_SIZE_LIMIT_IN_MB || "3", 10) * 1024 * 1024;
+
+const client = new LambdaClient({
+  region: process.env.AWS_REGION || "us-east-1"
+});
+
+const waiterConfig = {
+  client,
+  maxWaitTime: MAX_WAIT_TIME,
+  minDelay: DELAY,
+  maxDelay: DELAY
+};
+
+const createZip = async (fileName, code) => {
+  return new Promise((resolve, reject) => {
+    const zip = new JSZip();
+    zip.file("transform_wrapper.py", TRANSFORM_WRAPPER_CODE);
+    zip.file("user_transformation.py", code);
+
+    zip
+      .generateNodeStream({ type: "nodebuffer", streamFiles: true })
+      .pipe(fs.createWriteStream(`${fileName}.zip`))
+      .on("finish", () => {
+        logger.debug(`${fileName}.zip has been created`);
+        resolve();
+      })
+      .on("error", err => {
+        stats.counter("create_zip_error", 1, { fileName });
+        logger.error(`Error while creating zip file: ${err.message}`);
+        reject(err);
+      });
+  });
+};
+
+const createLambdaFunction = async (functionName, code, publish) => {
+  try {
+    logger.debug(`Creating lambda for ${functionName} with publish:${publish}`);
+    // create, read and delete zip
+    await createZip(functionName, code);
+    const zipContent = fs.readFileSync(`./${functionName}.zip`);
```

Comment:
should we use async version from fs/promises

Reply:
`fs.readFileSync` is `await fs.readFile<Promise>`
is there any reason behind to move to promises ?

Reply:
Otherwise, it will be a blocking op on the Node processor.